src/common/Utilities/Util.cpp

```diff
@@ -541,19 +544,26 @@ void utf8printf(FILE* out, const char *str, ...)
 void vutf8printf(FILE* out, const char *str, va_list* ap)
 {
 #if AC_PLATFORM == AC_PLATFORM_WINDOWS
-    char temp_buf[32 * 1024];
-    wchar_t wtemp_buf[32 * 1024];
+    std::string temp_buf;
+    temp_buf.resize(32 * 1024);
+    std::wstring wtemp_buf;
 
-    size_t temp_len = vsnprintf(temp_buf, 32 * 1024, str, *ap);
-    //vsnprintf returns -1 if the buffer is too small
-    if (temp_len == size_t(-1))
-        temp_len = 32*1024-1;
+    std::size_t temp_len = vsnprintf(&temp_buf[0], 32 * 1024, str, *ap);
+    temp_buf.resize(strnlen_s(temp_buf.c_str())); // Resize to match the formatted string
 
-    size_t wtemp_len = 32*1024-1;
-    Utf8toWStr(temp_buf, temp_len, wtemp_buf, wtemp_len);
+    if (!temp_buf.empty())
+    {
+        Utf8toWStr(temp_buf, wtemp_buf, 32 * 1024);
+        wtemp_buf.push_back('\0');
+
```

Comment:
There might be a parenthesis missing here，

Reply:
???

Reply:
ah i see lol )

Reply:
but this is not need for #2748 you need only src/server/game/Handlers/MailHandler.cpp changes